src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -261,14 +261,24 @@ inline void JavaThread::set_done_attaching_via_jni() {
 
 inline bool JavaThread::is_exiting() const {
   // Use load-acquire so that setting of _terminated by
-  // JavaThread::exit() is seen more quickly.
+  // JavaThread::set_terminated() is seen more quickly.
```

Comment:
This comment should have been updated when the code in
JavaThread::exit() was refactored into JavaThread::set_terminated().
I'm doing it as part of this fix for clarity.

Reply:
I'm pretty sure that I wrote the original comment and you were one of my
reviewers way, way back then.

When we were trying to match up the load-acquires in the accessors with
the in-line release-store code in JavaThread::exit(), the comment was needed
(ignoring the part about "seeing things more quickly"). Now that we have
load-acquires in the accessors and a release-store in the setter, i.e., no more
in-line code in JavaThread::exit(), I no longer think we need a comment to
coordinate.

I'm in favor of dropping all instances of this comment.

Reply:
Sounds good. FTR this came in with the Thread-SMR changes which were reviewed over a period of many months with some "vigorous" discussion. :) I think by the time this particular code appeared the focus was on bigger issues.